app/gui/src/beforeMain.ts

```diff
@@ -0,0 +1,49 @@
+import * as detect from 'enso-common/src/detect'
+
+if (detect.IS_DEV_MODE) {
+    suppressReactAriaConsoleWarnings()
+    suppressVueDevToolsConsoleWarnings()
+}
+
+function suppressConsoleMessage(
+    message: string | RegExp | (string | RegExp)[] | ((...args: unknown[]) => boolean),
+    level: 'warn' | 'error' | 'log' | 'debug' | 'info' = 'warn',
+  ) {
+    const originalConsoleMethod = console[level]
+  
+    console[level] = function overrideConsoleMethod(...args: unknown[]) {
+      let shouldSuppress = false
+  
+      switch (true) {
+        case typeof message === 'function':
+          shouldSuppress = message(...args)
+          break
+        case typeof message === 'string':
+          shouldSuppress = args[0] === message
+          break
+        case Array.isArray(message):
+          shouldSuppress = message.some((m) =>
+            typeof m === 'string' ? args[0] === m : m.test(args[0] as string),
+          )
+          break
+        default:
+          shouldSuppress = message.test(args[0] as string)
+          break
+      }
+  
+      if (shouldSuppress) {
+        return
+      }
+  
+      return originalConsoleMethod.apply(console, args)
+    }
+  }
+  
+  function suppressReactAriaConsoleWarnings() {
+    suppressConsoleMessage(/A PressResponder was rendered without a pressable child/)
+    suppressConsoleMessage(/Download the React DevTools for a better development experience/, 'info')
```

Comment:
This is really the only added line. The file is separated into separate module in order to guarantee that it runs before other imported scripts, thus giving us a way to suppress that console message (otherwise it would be logged before suppression code executes).